### PR TITLE
Adding instructions to run cluster api tilt environment

### PR DESCRIPTION
**Pull request by jan-est**
This PR adds step-by-step documents to set up Kind cluster and use Tilt for Cluster API development and testing.

### CODE REVIEW

File: capi-tilt/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+kubectl --namespace=default get secret/my-cluster-kubeconfig -o json | jq -r .data.value | base64 --decode   > ./capi-quickstart.kubeconfig
+```
+
+## Install and patch Calico 
+
+In this example we deploy Calico CNI solution. Deploy Calico using **capi-quickstart.kubeconfig** copied in last step with:
+
+```
+kubectl --kubeconfig=./capi-quickstart.kubeconfig \
+  apply -f https://docs.projectcalico.org/v3.8/manifests/calico.yaml
```
Comment (Jaakko-Os) [inline capi-tilt/README.md]: v3.8 is rather old.
have you tried v3.11 or v3.12?

Comment (jan-est) [reply capi-tilt/README.md]: No I have not tried it. These lines is taken from: 

https://master.cluster-api.sigs.k8s.io/user/quick-start.html

and these steps follows cluster api books own workflow.

File: capi-tilt/cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+    serviceDomain: "cluster.local"
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: DockerCluster
+    name: my-cluster
+    namespace: default
```
Comment (maelk) [inline capi-tilt/cluster.yaml]: Can you make it use CACPK ? You can see an example in metal3-dev-env

Comment (jan-est) [reply capi-tilt/cluster.yaml]: Could you be more specific? Don't know exactly what to look for?

Comment (jan-est) [reply capi-tilt/cluster.yaml]: Unfortunately it seems that CAPD does not support control plane management. [#252](https://github.com/kubernetes-sigs/cluster-api-provider-docker/issues/252)

Comment (jan-est) [reply capi-tilt/cluster.yaml]: got this working with CACPK

File: capi-tilt/README.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+'
+
+```
+
+## Create machinedeployment
+
+Finishing up the test cluster, create 3 node machinedeployment with:
+
+```
+kubectl apply -f machine-deployment.yaml
```
Comment (Xenwar) [inline capi-tilt/README.md]: change to
machine-deployments.yaml

**Comment by kashifest** [comment]
@jan-est can you make this PR towards the already existing CAPI-dev folder as that folder which I created earlier seems to be obsolete now according to your tests. So I don't see any point of keeping old information.

**Comment by jan-est** [comment]
@kashifest if it is OK to you I will delete the old CAPI-dev folder with my PR?

**Comment by kashifest** [comment]
> @kashifest if it is OK to you I will delete the old CAPI-dev folder with my PR?

Sure. If nothing is needed from that folder , go ahead and delete it.

File: capi-tilt/README.md (hunk 3 above)
Comment (jan-est) [reply capi-tilt/README.md]: I changed the yaml name to machine-deployment.yaml

**Comment by Jaakko-Os** [comment]
looks ok to me

**Pull request merged**